### ISSUE TITLE
Write checkpointing proof-of-concept

### ISSUE THREAD

**Issue by borjasotomayor** [uchicago-cs/chiventure#14]
For the 2-3 formats decided in #13, write a proof-of-concept program that (1) saves some game data to a file, and (2) loads it back in.

**Comment by chrisfernandes18** [comment]
I looked into the msgpack library and constructed a game struct which included a 2D array, the column value, the row value, a string description, and an enum for player. I used msgpack to then serialize and export an example game struct into a file, which is in a module called save.c. Then I created a load.c module which reads the file, de-serializes the file, and loads it back into an array. 

Overall, using msgpack was a pain because there is little documentation for the library in C. Further, the only documentation was really a list of all the functions, an example implementation which used an obsolete function. It was difficult finding information on Stack Overflow as well, but I was able to get it to work so I guess it's okay.

**Comment by schin99** [comment]
I used protobuf-c and constructed a game that would save a Player (name and gender), the Room they would have saved in, and an int that would indicated the number of rooms the player had found. Similar to Christian, I used protobuf-c to serialize and export a sample game into a file and deserialize it using modules also called save.c and load.c.

Compared to msgpack, protobuf didn't seem too difficult to use. The documentation for protobuf wasn't too difficult although the examples that I could use were a little sparse. I did have some trouble understanding everything, but it ended up being much simpler than I thought.
